### PR TITLE
Compaction documentation follow-up

### PR DESCRIPTION
**Pull request by lucperkins**
PR #1466 was merged before I was able to address a second round of comments from @ivankelly. This PR addresses those comments.

Ready for review.

### CODE REVIEW

**Comment by sijie** [comment]
@ivankelly please review it

**Comment by sijie** [comment]
@ivankelly can you review this?

File: site/docs/latest/cookbooks/compaction.md [hunk 1]
```diff
@@ -11,13 +11,18 @@ To use compaction:
 * You must manually [trigger](#trigger) compaction using the Pulsar administrative API. This will both run a compaction operation *and* mark the topic as a compacted topic.
 * Your {% popover consumers %} must be [configured](#config) to read from compacted topics ([Java consumers](#java), for example, have a `readCompacted` setting that must be set to `true`). If this configuration is not set, consumers will still be able to read from the non-compacted topic.
 
+{% include admonition.html type="warning" content="Compaction only works on messages that have keys (as in the stock ticker example the stock symbol serves as the key for each message). Keys can thus be thought of as the axis along which compaction is applied. Messages that don't have keys are simply ignored by compaction." %}
```
Comment (ivankelly) [inline site/docs/latest/cookbooks/compaction.md]: "simply ignored" can be confusing. Does it mean it just throws them away or passes them through? (it passes them through).

**Comment by sijie** [comment]
@ivankelly let's merge this and can you fix the comment. so we can move it forward with 2.1 release.

**Comment by sijie** [comment]
@ivankelly can you check my [last comment](https://github.com/apache/incubator-pulsar/pull/1871#issuecomment-396444300)?

**Comment by sijie** [comment]
retest this please

**Pull request merged**